### PR TITLE
gh-91217: deprecate imghdr

### PR DESCRIPTION
**Pull request by brettcannon**
Automerge-Triggered-By: GH:warsaw

### CODE REVIEW

File: Doc/whatsnew/3.11.rst [hunk 1]
```diff
@@ -852,9 +852,14 @@ Deprecated
...
   * :mod:`chunk`
   * :mod:`crypt`
+  * :mod:`imghdr`
 
   (Contributed by Brett Cannon in :issue:`47061`.)
 
+* The default argument for *_subtype* for :class:`email.mime.MIMEImage`
+  has been deprecated due to the deprecation of :mod:`imghdr`. Please
+  explicitly provide a an image subtype.
+
```
Comment (hugovk) [inline Doc/whatsnew/3.11.rst]: Include reference?

```suggestion
  explicitly provide a an image subtype.

(Contributed by Brett Cannon in :issue:`47061`.)
```

File: Doc/library/email.mime.rst [hunk 2]
```diff
@@ -204,6 +204,10 @@ Here are the classes:
    .. versionchanged:: 3.6
       Added *policy* keyword-only parameter.
 
+   .. deprecated:: 3.11
+      The default argument for *_subtype* due :mod:`imghdr` being
```
Comment (hugovk) [inline Doc/library/email.mime.rst]: Not sure if this is a typo or just different flavours of English :)

```suggestion
      The default argument for *_subtype* due to :mod:`imghdr` being
```

File: Lib/email/mime/image.py [hunk 3]
```diff
@@ -38,6 +41,12 @@ def __init__(self, _imagedata, _subtype=None,
         header.
         """
         if _subtype is None:
+            fullname = f"{__name__}.{self.__class__.__qualname__}"
+            msg = ("The *_subtype* argument's default value for {name}"
+                   "is deprecated due to the deprecation of imghdr (slated for "
+                   "removal in Python {remove}); explicitly specify the image "
+                   "subtype")
```
Comment (hugovk) [inline Lib/email/mime/image.py]: This is telling people who _aren't_ using `_subtype`, to start using this "private" argument.

Should it be promoted to a "public" argument? Or not worth the effort?

Comment (brettcannon) [reply Lib/email/mime/image.py]: It's not private based on the documentation explicitly referencing it. I think this is part of how the `email` package exposes the class via some other mechanisms.

File: Doc/includes/email-mime.py [hunk 4]
```diff
@@ -22,7 +19,7 @@
     with open(file, 'rb') as fp:
         img_data = fp.read()
     msg.add_attachment(img_data, maintype='image',
-                                 subtype=imghdr.what(None, img_data))
+                                 subtype="jpeg")
```
Comment (hugovk) [inline Doc/includes/email-mime.py]: Minor quibble: we have a mix of single and double quotes here

**Review by hugovk — APPROVED** [review]

File: Doc/includes/email-mime.py (hunk 4 above)
Comment (warsaw) [reply Doc/includes/email-mime.py]: I was going to try to edit this to switch it to single quotes for consistency, but it doesn't look like I can do that in the web ui.

File: Doc/library/email.mime.rst [hunk 5]
```diff
@@ -180,11 +180,11 @@ Here are the classes:
    A subclass of :class:`~email.mime.nonmultipart.MIMENonMultipart`, the
    :class:`MIMEImage` class is used to create MIME message objects of major type
    :mimetype:`image`. *_imagedata* is a string containing the raw image data.  If
-   this data can be decoded by the standard Python module :mod:`imghdr`, then the
-   subtype will be automatically included in the :mailheader:`Content-Type` header.
-   Otherwise you can explicitly specify the image subtype via the *_subtype*
-   argument.  If the minor type could not be guessed and *_subtype* was not given,
-   then :exc:`TypeError` is raised.
+   this data can be decoded by the standard Python module :mod:`imghdr`
+   (which is deprecated), then the subtype will be automatically included in the
```
Comment (warsaw) [inline Doc/library/email.mime.rst]: Why don't we just inline that into the email package and either expose it as a public API or just leave it private?

**Review by warsaw — CHANGES_REQUESTED** [review]

File: Doc/includes/email-mime.py (hunk 4 above)
Comment (hugovk) [reply Doc/includes/email-mime.py]: Can you check https://github.com/python/cpython/pull/91461/files and then the ellipsis at the top right by the filename should have an "Edit file" link?

File: Doc/library/email.mime.rst (hunk 5 above)
Comment (warsaw) [reply Doc/library/email.mime.rst]: I was probably misreading the code before, but the question still remains whether we should inline `imghdr` into the email package.

File: Doc/includes/email-mime.py (hunk 4 above)
Comment (warsaw) [reply Doc/includes/email-mime.py]: Thanks @hugovk that did the trick.

File: Doc/library/email.mime.rst (hunk 5 above)
Comment (brettcannon) [reply Doc/library/email.mime.rst]: If we did that how would you document that? Document all file formats in the constructor's docs?

**Comment by brettcannon** [comment]
I have made the requested changes; please review again.

**Review by warsaw — APPROVED** [review]
LGTM.  I'd like to do a cleaning pass through the inlined code, but I'll do that in a follow up PR.

File: Doc/includes/email-mime.py [hunk 6]
```diff
@@ -22,7 +19,7 @@
     with open(file, 'rb') as fp:
         img_data = fp.read()
     msg.add_attachment(img_data, maintype='image',
-                                 subtype=imghdr.what(None, img_data))
+                                 subtype='jpeg')
```
Comment (ngnpope) [inline Doc/includes/email-mime.py:22]:
```suggestion
                                 subtype='png')
```

A few lines above there is `for file in pngfiles:`…

Comment (warsaw) [reply Doc/includes/email-mime.py:22]: Since this PR is already merged, I'll add this to my clean up branch at #91520

Comment (ngnpope) [reply Doc/includes/email-mime.py:22]: 👍🏻

**Pull request merged**